mods/sysdm-general-tab.wh.cpp

```diff
@@ -0,0 +1,1298 @@
+    GetModuleFileNameW(GetModuleHandle(0), szLoadedModule, sizeof(szLoadedModule));
+    Wh_Log(L"Loaded into module: %s", szLoadedModule);
+
+    if (StrStrI(szLoadedModule, L"\\control.exe") != nullptr)
+    {
+        // We're loading into control.exe, so do hack to see what control.exe is trying
+        // to bring up and take over if we need to. Otherwise, bail out - we have no
+        // work to do in a control.exe instance other than redirection.
+        LPCWSTR szCommandLine = GetCommandLineW();
+        if (StrStrI(szCommandLine, L"sysdm.cpl") != nullptr)
+        {
+            Wh_Log(L"Opening sysdm.cpl from control.exe.");
+
+            // control.exe is hardcoded to open SystemPropertiesComputerName.exe from the
+            // system folder in Windows 10. We'll repeat this behaviour, but pass an
+            // argument so we know to change the default tab.
+            WCHAR szPath[MAX_PATH];
+            GetSystemDirectoryW(szPath, ARRAYSIZE(szPath));
+            StringCchCatW(szPath, ARRAYSIZE(szPath), L"\\SystemPropertiesComputerName.exe");
+            ShellExecuteW(
+                nullptr,
+                L"open",
+                szPath,
+                L"/opengeneraltab",
+                nullptr,
+                SW_SHOW
+            );
+
+            TerminateProcess(GetCurrentProcess(), 0);
+            return TRUE;
+        }
+
+        return FALSE;
+    }
+
+    g_hInst = GetModuleHandleW(L"sysdm.cpl");
+
+    if (!g_hInst)
+    {
+        Wh_Log(L"For some reason, sysdm.cpl isn't loaded. Bailing!");
+        return FALSE;
+    }
+
+    LPCWSTR szCommandLine = GetCommandLineW();
+    if (StrStrIW(szCommandLine, L"/opengeneraltab"))
+    {
+        g_fHijackDefaultTab = true;
+    }
+
+    LoadSettings();
+
+    g_hInstResource = LoadLibraryExW(g_settings.pszXpSysdm, nullptr, LOAD_LIBRARY_AS_IMAGE_RESOURCE);
+
+    // Get the version of the resource DLL to determine which version of Windows
+    // it is from:
+    DWORD dwVersionSize = GetFileVersionInfoSizeW(g_settings.pszXpSysdm, nullptr);
+    BYTE *pbVersionInfo = new BYTE[dwVersionSize];
+    VS_FIXEDFILEINFO *pFileInfo;
+    UINT uLenFileInfo = 0;
+    if (GetFileVersionInfoW(g_settings.pszXpSysdm, 0, dwVersionSize, pbVersionInfo))
+    {
+        if (VerQueryValueW(pbVersionInfo, L"\\", (void **)&pFileInfo, &uLenFileInfo))
+        {
+            g_dwResourceVersion = pFileInfo->dwFileVersionMS;
+        }
+        else
+        {
+            Wh_Log(L"Query for file version information failed.");
+        }
+    }
+    else
+    {
+        Wh_Log(L"Failed to get file version size.");
+    }
+    delete[] pbVersionInfo;
+
+    WindhawkUtils::Wh_SetFunctionHookT(
+        PropertySheetW,
+        PropertySheetW_hook,
+        &PropertySheetW_orig
+    );
+
+    return TRUE;
+}
+
+// The mod is being unloaded, free all allocated resources.
+void Wh_ModUninit()
+{
+    Wh_Log(L"Uninit");
+    Wh_FreeStringSetting(g_settings.pszXpSysdm);
```

Comment:
What about the other strings? Or you can just use `WindhawkUtils::StringSetting`, for example:
https://github.com/ramensoftware/windhawk-mods/blob/37479b14fd4a35d62fdb2fec9d9e0d66f8978c57/mods/custom-shutdown-dialog.wh.cpp#L91